### DISCUSSION TITLE
juicefs sync failed if key contains "/"

### DISCUSSION THREAD

**Discussion by n0v3xx** [juicedata/juicefs#873] [General]
**What happened**:
juicefs sync fails because a / is in the secret key.

**What you expected to happen**:
the bucket key should contains any char and juicefs sync should work!

**How to reproduce it (as minimally and precisely as possible)**:
Try to sync something with a bucket that contains a "/" in the secret key.

**Anything else we need to know?**
Can't parse s3://*****************:******************/*****@bucket/juicefs-backup/backup_juicefs.json: parse "s3://*****************:******************@bucket/juicefs-backup/backup_juicefs.json": invalid port ":******************" after host

**Environment**:
- JuiceFS version (use `./juicefs --version`) or Hadoop Java SDK version:  juicefs version 0.17.0 (2021-09-24T04:17:26Z e115dc4)
- Cloud provider or hardware configuration running JuiceFS: b2
- OS (e.g: `cat /etc/os-release`): ubuntu 20
- Kernel (e.g. `uname -a`):
- Object storage (cloud provider and region): b2
- Redis info (version, cloud provider managed or self maintained):
- Network connectivity (JuiceFS to Redis, JuiceFS to object storage):
- Others:

**Comment by davies** [comment]
Can you try to replace the `/` as `%2F`?

**Comment by n0v3xx** [comment]
That works, but its a unexpected behavior. Why does the end user have to fix this? This can be fixed by juicefs.

**Comment by davies** [comment]
We expect the `bucket` to be a valid URL, but when there is a `/` in access key or secret key, the string is not a valid URL anymore, we can't parse the string correctly. That's a workaround we have to use unfortunately.
